### ISSUE TITLE
Geth returns 0 balance, bug? 

### ISSUE THREAD

**Issue by teb** [ethereum/go-ethereum#16692]
Hi! ,
Running Geth 1.7.3 -> now 1.8.7 im getting:

```
eth.getBlock("latest").number = 0
web3.fromWei(eth.getBalance(eth.coinbase), "ether") = 0
 eth.syncing
{
  currentBlock: 5571499,
  highestBlock: 5571593,
  knownStates: 65815495,
  pulledStates: 65804256,
  startingBlock: 5569311
```

LOG:
```
INFO [05-07|12:56:43] Imported new state entries               count=1070 elapsed=6.021ms   processed=66261502 pending=9851  retry=0   duplicate=88 unexpected=368
INFO [05-07|12:56:43] Imported new state entries               count=988  elapsed=5.010ms   processed=66262490 pending=9718  retry=82  duplicate=88 unexpected=368
INFO [05-07|12:56:43] Imported new state entries               count=1070 elapsed=6.012ms   processed=66263560 pending=9205  retry=0   duplicate=88 unexpected=368
INFO [05-07|12:56:45] Imported new state entries               count=987  elapsed=3.016ms   processed=66264547 pending=11618 retry=83  duplicate=88 unexpected=368
INFO [05-07|12:56:46] Imported new state entries               count=768  elapsed=3.798ms   processed=66265315 pending=13863 retry=0   duplicate=88 unexpected=368
```

The last time i tried this in february i got 2,5 ETH, and ethermine clearly states i have 2.5 eth.
My last payout was around block : 5035500 (im synced way past that)

#### System information
OS: Windows 10x64 (ryzen 8 core, 16gb ram)
Geth: 1.8.7 (running on a m2 SSD)

Any tips are mostly welcomed 

/D

**Comment by holiman** [comment]
> im synced way past that

Fast-synching doesn't work like that, until you have _all_ entries, there are no guarantees that the data is correct. If it were a full-sync, that statement would have been true. However, fast-sync fetches the blocks pretty quickly, but fetching the gigantic state takes a lot more time to complete.

**Comment by holiman** [comment]
> Any tips are mostly welcomed 

So either check you balance on etherscan.io, or use LES (`--light`) to quickly get the info you want.

**Comment by teb** [comment]
thx for the response.
1. Since i started to mine i was able to check my balance til feb 2018, (i had a pause for 2 months) and now i cant.. what changed? (i used the same syntax:  geth --maxpeers 25 --cache 64000 -datadir "s:\ethereum""
2. The problem is that i cant transfer my ETH to another wallet without GETH beeing in sync
3. From my oppinion, it never seems to get ALL entries.. Even on a 200mbps link, M2 SSD etc..
Im not sure what to do.. have i lost the eth i mined?

**Comment by tahder** [comment]
I also got the same issue, I got 20 but when I restart as not sync well, goes to 0.
![image](https://user-images.githubusercontent.com/5017393/39978802-b99e2582-5796-11e8-9b16-a80963f1799b.png). At  etherscan.io displays 0 and no logs or transactions logs.

which I used this command, `geth --fast --cache=2048 --mine --minerthreads=4` 

Currently I am now removing the --fast, hope someone give an insights as newbie with this ethereum. Or what I miss? or really a bug?

Btw I am using Fedora 27 with 
` geth version
`Geth
Version: 1.8.7-stable
Git Commit: 66432f3821badf24d526f2d9205f36c0543219de
Architecture: amd64
Protocol Versions: [63 62]
Network Id: 1
Go Version: go1.10.1
Operating System: linux
